_faqs/01-01-what-is-opensearch.markdown

```diff
@@ -1,6 +1,7 @@
 ---
 question: What is OpenSearch?
 category: General
+ref: q1.1
```

Comment:
@dblock I don't think using frontmatters to assign the ID is the right take. I had initially considered this method but I went with the auto-assign for two reasons:
1) explicit ID references run the risk of duplicate IDs (breaking linking, also bad practice)
2) this requires some knowledge of what `ref` does - so for folks doing drive-by contributions, it requires a deeper understanding. The filename is more explicit to those who aren't deeply involved. 

I'll admit using filenames as a sorting mechanism seems janky, but I think it's the best option.

Reply:
It's not just janky, it's broken by design without warning. For example, introducing a new category will sort the list of categories differently and renumber everything. You're sure?